src/Common/tests/System/Net/Configuration.Certificates.cs

```diff
@@ -45,14 +45,24 @@ private static X509Certificate2Collection GetCertificateCollection(string certif
             {
                 // On Windows, .Net Core applications should not import PFX files in parallel to avoid a known system-level race condition.
                 // This bug results in corrupting the X509Certificate2 certificate state.
-                Assert.True(m.WaitOne(MutexTimeout), "Cannot acquire the global certificate mutex.");
-
-                var certCollection = new X509Certificate2Collection();
-                certCollection.Import(Path.Combine(TestDataFolder, certificateFileName), CertificatePassword, X509KeyStorageFlags.DefaultKeySet);
+                try
+                {
+                    Assert.True(m.WaitOne(MutexTimeout), "Cannot acquire the global certificate mutex.");
 
-                m.ReleaseMutex();
+                    var certCollection = new X509Certificate2Collection();
+                    certCollection.Import(Path.Combine(TestDataFolder, certificateFileName), CertificatePassword, X509KeyStorageFlags.DefaultKeySet);
```

Comment:
> I assume you're talking about the finally block, it seems odd to me that it can run before the catch block finishes executing

No, I'm talking about the mutex finalizer (not the `finally` block). 
I get the AbandonedMutexException only when the process is killed via something like Task Manager.

MSDN is not clear on the semantics. The Albahari C# in a Nutshell book is saying that "Closing or disposing a Mutex automatically releases it". A normal application shutdown will run all finalizers of Disposable objects.

This seems to be in-line with the behavior that I'm observing: a "normal" application termination will not hold the mutex even if ReleaseMutex is not called.